src/components/Dashboard/index.js

```diff
@@ -73,283 +127,488 @@ const Dashboard = () => {
+  const validateOrgHandle = async () => {
     const orgHandleExists = await checkOrgHandleExists(orgHandle)(
       firebase,
       dispatch
     );
+    if (validator.isEmpty(orgHandle)) {
+      setOrgHandleValidateError(true);
+      setOrgHandleValidateErrorMessage("Please enter a handle");
+      return false;
+    } else if (
+      !validator.isAlphanumeric(orgHandle) ||
+      !validator.isLowercase(orgHandle)
+    ) {
+      setOrgHandleValidateError(true);
+      setOrgHandleValidateErrorMessage(
+        "Organization handle can only contain lowercase alphanumeric characters"
+      );
+      return false;
+    } else if (orgHandle.length < 6) {
+      setOrgHandleValidateError(true);
+      setOrgHandleValidateErrorMessage(
+        "Organization handle cannot be less than 6 characters"
+      );
+      return false;
+    } else if (orgHandleExists) {
+      setOrgHandleValidateError(true);
+      setOrgHandleValidateErrorMessage(
+        `The handle ${orgHandle} is already taken`
+      );
+      return false;
+    } else {
+      setOrgHandleValidateError(false);
+      setOrgHandleValidateErrorMessage("");
+      return true;
+    }
+  };
+
+  const onChangeCountry = (country) => setCountry(country);
+  const validateCountry = () => {
+    console.log(country);
+    if (validator.isEmpty(country)) {
+      setCountryValidateError(true);
+
+      return false;
+    } else {
+      setCountryValidateError(false);
 
-    if (orgHandleExists) {
-      form.resetFields(["org_handle"]);
-      form.setFields([
-        {
-          name: "org_handle",
-          errors: [`The handle [${orgHandle}] is already taken`],
-        },
-      ]);
+      return true;
     }
   };
 
+  const onChangeOrgCountry = (orgCountry) => setOrgCountry(orgCountry);
+  const validateOrgCountry = () => {
+    console.log(orgCountry);
+    if (validator.isEmpty(orgCountry)) {
+      setOrgCountryValidateError(true);
+
+      return false;
+    } else {
+      setOrgCountryValidateError(false);
+
+      return true;
+    }
+  };
+
+  const onChangeOrgWebsite = (orgWebsite) => setOrgWebsite(orgWebsite);
+  const validateOrgWebsite = () => {
+    if (validator.isEmpty(orgWebsite)) {
+      setOrgWebsiteValidateError(true);
+      setOrgWebsiteValidateErrorMessage("Please enter organization website");
+      return false;
+    } else if (!validator.isURL(orgWebsite)) {
+      setOrgWebsiteValidateError(true);
+      setOrgWebsiteValidateErrorMessage("Please provide a valid URL");
+      return false;
+    } else if (
+      !(orgWebsite.includes("https://") || orgWebsite.includes("http://"))
+    ) {
+      setOrgWebsiteValidateError(true);
+      setOrgWebsiteValidateErrorMessage(
+        "URL must contain the protocol (https:// or http://)"
+      );
+      return false;
+    } else {
+      setOrgWebsiteValidateError(false);
+      setOrgWebsiteValidateErrorMessage("");
+      return true;
+    }
+  };
```

Comment:
Can we look into refactoring these validation functions into a separate file? In that way, we will be able to reuse them in other components as well while keeping the size of this component small.

Reply:
I will add validations to a shared file. I'm currently working on that. Thank you for mentioning that.